### PR TITLE
PP-1828 Reorganised README.md for better readability, added new sections and new CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ckalista**
## WHAT
_A brief description of the pull request:_

- Reorganised README.md:
   - Extracted out the API specification into its own file for readability.
   - Added licence section.
   - Added responsible disclosure section.

- Created new CONTRIBUTING.md

## HOW 
_Steps to test or reproduce:_

### CODE REVIEW

**Comment by ghost** [comment]
There were no errors, go have a cup of coffee...

File: doc/api_specification.md [hunk 1]
```diff
@@ -0,0 +1,1579 @@
+# API Specification
```
Comment (markyjones) [inline doc/api_specification.md]: is it OK to move doc/api_specification.md into a docs folder? i.e. docs/api_specification.md? Just to keep it consistent with the other repros? I would change myself but have a meeting in 2mins(!)

File: README.md [hunk 2]
```diff
@@ -59,1623 +59,57 @@ The command to run all the tests is:
 ```
     mvn test
 ```
-## TASKS NAMESPACE
 
-| Path                          | Supported Methods | Description                        |
-| ----------------------------- | ----------------- | ---------------------------------- |
-|[```/v1/tasks/expired-charges-sweep```](#post-v1tasksexpired-charges-sweep)  | POST    |  Spawns a task to expire charges with a default window of 1 Hr |   
+## API Specification
```
Comment (markyjones) [inline README.md:63]: I have been leading this section with the following statement with a link to the spec.

> The API Specification provides more detail on the paths and operations including examples.

e.g. https://github.com/alphagov/pay-publicauth/blob/master/README.md#api-specification

File: doc/api_specification.md (hunk 1 above)
Comment (ckalista) [reply doc/api_specification.md]: Done, I thought of doing it but weren't sure wether the `doc` folder was already being referenced anywhere else.

File: README.md (hunk 2 above)
Comment (ckalista) [reply README.md:63]: Completely missed that - thanks for spotting.

**Comment by ghost** [comment]
There were no errors, go have a cup of coffee...

**Review by markyjones — APPROVED** [review]
Thanks for making the changes as discussed.

**Pull request merged**